pallets/rmrk-core/src/lib.rs

```diff
@@ -395,6 +399,21 @@ pub mod pallet {
 			}
 			.unwrap_or_default();
 
+			let (root_owner, root_nft) = Pallet::<T>::lookup_root_owner(collection_id, nft_id)?;
+			// Check ownership
+			ensure!(sender == root_owner, Error::<T>::NoPermission);
+			// Prepare transfer
+			let new_owner_account = match new_owner.clone() {
+				AccountIdOrCollectionNftTuple::AccountId(id) => id,
+				AccountIdOrCollectionNftTuple::CollectionAndNftTuple(cid, nid) => {
+					// Circle detection
+					let (_, new_owner_root_nft) = Pallet::<T>::lookup_root_owner(cid, nid)?;
+					ensure!(root_nft != new_owner_root_nft, Error::<T>::CircleDetected);
```

Comment:
Oops, I think this is not correct (so does the code in my issue). The purpose is to ensure the NFT will not send to its descendant, because otherwise it will form a cycle. However, the current implementation also forbids sending to another branch in the current tree.

Example below. If we move 2 to 2' (sending 2 to 1). This operation is allowed in the spec, but it won't pass this check.

```
                           
     root                  
      /\                   
     /  \                  
   +-+  +-+                
   |1|  |2|                
   +-+  +-+                
    /                      
   /                       
+---+                      
| 2'|                      
+---+     
```

This is the situation we really want to prevent. If we send 2 to 3, it will form a cycle.

```
                          
    root                  
     /\                   
    /  \                  
  +-+  +-+                
  |1|  |2|                
  +-+  +-+                
        \                 
         \                
        +-+               
        |3|               
        +-+               
         /                
        /                 
     +---+                
     |2'!|                
     +---+   
```